runtime/src/main/java/org/corfudb/protocols/CorfuProtocolLogData.java

```diff
@@ -0,0 +1,59 @@
+package org.corfudb.protocols;
+
+import com.google.protobuf.ByteString;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.runtime.proto.LogData.LogDataMsg;
+import org.corfudb.runtime.proto.LogData.ReadResponseMsg;
+
+/**
+ * This class provides methods for creating and converting between the Protobuf
+ * objects defined in log_data.proto and their Java counterparts. Used by the
+ * LogUnit RPCs.
+ */
+@Slf4j
+public class CorfuProtocolLogData {
+    // Prevent class from being instantiated
+    private CorfuProtocolLogData() {}
+
+    /**
+     * Returns the Protobuf representation of a LogData object.
+     *
+     * @param logData  the desired LogData object
+     * @return         an equivalent Protobuf LogData message
+     */
+    public static LogDataMsg getLogDataMsg(LogData logData) {
+        ByteBuf buf = Unpooled.buffer();
+        logData.doSerialize(buf);
+
+        return LogDataMsg.newBuilder()
+                .setEntry(ByteString.copyFrom(buf.resetReaderIndex().array()))
+                .build();
+    }
+
+    /**
+     * Returns a LogData object from its Protobuf representation.
+     *
+     * @param msg  the desired Protobuf LogData message
+     * @return     a equivalent LogData object
+     */
+    public static LogData getLogData(LogDataMsg msg) {
+        return new LogData(Unpooled.wrappedBuffer(msg.getEntry().toByteArray()));
+    }
```

Comment:
I think `new LogData(Unpooled.wrappedBuffer(msg.getEntry().asReadOnlyByteBuffer()));` is cheaper than `toByteArray` since it will eliminate one allocation and one copy.